pageserver/src/tenant/timeline.rs

```diff
@@ -2964,6 +2967,29 @@ impl Timeline {
             .await?;
 
             keyspace.remove_overlapping_with(&completed);
+
+            // Do not descend into the ancestor timeline for aux files.
+            // We don't return a blanket [`GetVectoredError::MissingKey`] to avoid
+            // stalling compaction.
+            // TODO(chi): this will need to be updated for aux files v2 storage
+            if keyspace.overlaps(&NON_INHERITED_RANGE) {
```

Comment:
> NON_INHERITED_RANGE will be a keyspace containing multiple ranges in the future

Hmm. I assumed it would be a single range. Regardless, the same approach should work. The `overlaps` check on line 2975 is just an optimisation (`remove_overlapping_with` can be expensive). The check could simply be removed.

Reply:
as long as we fully disable aux v1 it will become a single range

Reply:
I'll leave this change up to you since I'm not sure what you'll need for aux v2.